test/e2e/specs/interactivity/router-regions.spec.ts

```diff
@@ -144,4 +198,125 @@ test.describe( 'Router regions', () => {
 		await expect( invalidRegionText1 ).toHaveText( 'content from page 1' );
 		await expect( invalidRegionText2 ).toHaveText( 'content from page 1' );
 	} );
+
+	test( 'should support router regions with the `attachTo` property.', async ( {
+		page,
+		interactivityUtils: utils,
+	} ) => {
+		await page.goto(
+			utils.getLink( 'router regions - page 1 - attachTo' )
+		);
+
+		const bodyLocator = page.locator( 'body' );
+		const regionsLocator = page.locator( '#regions-with-attach-to' );
+
+		const region3 = bodyLocator.getByTestId( 'region3' );
+		const region4 = regionsLocator.getByTestId( 'region4' );
+		const region5 = bodyLocator.getByTestId( 'region5' );
+		const region6 = regionsLocator.getByTestId( 'region6' );
+
+		const regions: Record< string, Locator > = {
+			region3,
+			region4,
+			region5,
+			region6,
+		};
+
+		// Regions with `attachTo` should initially be hidden.
+		await expect( region3 ).toBeHidden();
+		await expect( region4 ).toBeHidden();
+		await expect( region5 ).toBeHidden();
+		await expect( region6 ).toBeHidden();
+
+		// Navigate to "Page attachTo 1".
+		await page.getByTestId( 'next' ).click();
```

Comment:
Can we add an await here on the new page title to make sure that the navigation has finished before proceeding? (same for all the navigations)

Reply:
Done in https://github.com/woocommerce/gutenberg/pull/14/commits/af576295af2befccdee0e64a3a26c3eb92518509.